packages/orama/src/trees/avl.ts

```diff
@@ -53,29 +58,66 @@ function rotateRight<K, V>(node: Node<K, V>): Node<K, V> {
   return left
 }
 
-function findMin<K, V>(node: Node<K, V>): Node<K, V> {
-  return node.left ? findMin(node.left) : node
-}
-
 export function contains<K, V>(node: Node<K, V>, key: K): boolean {
   return !!find(node, key)
 }
 
-export function getSize<K, V>(node: Node<K, V> | null): number {
-  if (!node) {
-    return 0
+export function getSize<K, V>(root: Node<K, V> | null): number {
+  let size = 0;
+  const queue: Array<Node<K, V>> = [];
+
+  if (root !== null) {
+    queue.push(root);
   }
 
-  return 1 + getSize(node.left) + getSize(node.right)
+  while (queue.length > 0) {
+    const node = queue.shift() as Node<K, V>;
+    size++;
+
+    if (node.left !== null) {
+      queue.push(node.left);
+    }
+
+    if (node.right !== null) {
+      queue.push(node.right);
+    }
+  }
+
+  return size;
 }
 
-export function isBalanced<K, V>(node: Node<K, V> | null): boolean {
-  if (!node) {
-    return true
+export function isBalanced<K, V>(root: Node<K, V> | null): boolean {
+  const stack: StackNode<K, V>[] = [];
+
+  if (root !== null) {
+    stack.push({ node: root, checkedChildren: false });
   }
 
-  const heightDiff = Math.abs(getHeight(node.left) - getHeight(node.right))
-  return heightDiff <= 1 && isBalanced(node.left) && isBalanced(node.right)
+  while (stack.length > 0) {
+    const top = stack[stack.length - 1];
+
+    if (top.checkedChildren) {
+      const heightDiff = Math.abs(getHeight(top.node.left) - getHeight(top.node.right));
```

Comment:
I think it might be possible to skip the math.abs and optimize it a bit and just check if greater than 1 or smaller than -1 I need to think a bit more about this

Reply:
Removing `Math.abs` is slightly faster:

```
isBalancedOriginal x 3,987,396 ops/sec ±3.63% (87 runs sampled)
isBalancedOptimized x 4,018,579 ops/sec ±2.27% (85 runs sampled)
Fastest is isBalancedOptimized,isBalancedOriginal
```